### DISCUSSION TITLE
Install plugin with migrations should run cmd "kong migurations up" manually?

### DISCUSSION THREAD

**Discussion by SkyeBeFreeman** [Kong/kong#10300] [Help]
When should "kong migurations up" and "kong migurations finish" be executed?

**Comment by ghost** [comment]
Once all nodes have been upgraded to the current version of Kong Gateway, additional changes to the database can be made that are incompatible with the previous version.  To support that, each migration is split into two parts, an `up` part that can only make backwards-compatible changes, and a `teardown` part that runs after all nodes have been upgraded to the current version.

- `kong migrations up` is run to run the `up` part of all
   applicable migrations.
- `kong migrations finish` is run to invoke the `teardown` part
   of all applicable migrations.

**Reply by SkyeBeFreeman** [reply]
@sabertobihwy If my custom plugin contains `migrations`, should I execute `kong migrations up` and `kong migrations finish` manually?

**Reply by ghost** [reply]
> If my custom plugin contains `migrations`, should I execute `kong migrations up` and `kong migrations finish` manually?

I think so

**Reply by SkyeBeFreeman** [reply]
> > If my custom plugin contains `migrations`, should I execute `kong migrations up` and `kong migrations finish` manually?
> 
> I think so

I try to copy "jwt" plugin and install. Some migration tasks are in pending status and log shows "run kong migrations finish when ready", what does it mean? Does these migration tasks execute error? How can I check the reason of pending status?

**Reply by ghost** [reply]
> run kong migrations finish when ready

What is this log complete? I found 4 similar outputs in the code.

> I try to copy "jwt" plugin and install. Some migration tasks are in pending status and log shows

I think you'd better describe your procedure in detail so I can reproduce it.
